### ISSUE TITLE
South African Public Holidays Naming and Duplicates

### ISSUE THREAD

**Issue by surfer190** [peopledoc/workalendar#285]
### Setup

`workalendar==3.0.0`
`Python 3.6.4`

## Issue

The South Africa public holidays gives:

```
(datetime.date(2019, 1, 1), 'New year')
(datetime.date(2019, 3, 21), 'Human Rights Day')
(datetime.date(2019, 4, 19), 'Family Day')
(datetime.date(2019, 4, 19), 'Good Friday')
(datetime.date(2019, 4, 22), 'Easter Monday')
(datetime.date(2019, 4, 27), 'Freedom Day')
(datetime.date(2019, 5, 1), 'Workers Day')
(datetime.date(2019, 6, 16), 'Youth Day')
(datetime.date(2019, 6, 17), 'Youth Day substitute')
(datetime.date(2019, 8, 9), 'National Women Day')
(datetime.date(2019, 9, 24), 'Heritage Day')
(datetime.date(2019, 12, 16), 'Day of reconcilation')
(datetime.date(2019, 12, 25), 'Christmas Day')
(datetime.date(2019, 12, 26), 'Boxing Day')
(datetime.date(2019, 12, 26), 'Day of good will')
```

The problems:
* Family Day and Good Friday are on the same day. According to my source there should be a single name called **Good Friday**, the duplicate `Family Day should be removed`
* The is no Easter Monday anymore, it is called **Family Day**, this should be a rename.
* Boxing Day and Day of Goodwill are on the same day. According to my source there should be a single name called **Day of Goodwill**, the duplicate `Boxing Day should be removed`

You can view my [source of South African public holidays here](https://www.gov.za/about-sa/public-holidays)

**Comment by brunobord** [comment]
nice remarks!
I'm going to tackle this bug as soon as I can.

**Comment by brunobord** [comment]
as a reminder: https://en.wikipedia.org/wiki/Public_holidays_in_South_Africa

**Comment by surfer190** [comment]
Thanks for looking at this @brunobord 

I was thinking about it and wanted to know if it is meant for the package user to ensure a distinct holiday is on a specific day. Ie. if there are 2 or more on a single date, then package user need to ensure only first one is used for example. I assume the first item in the iterator will be the generic name.

I think the generic good friday, boxing day and easter monday are causing the duplicates. So perhaps we just leave it as is.

```
include_good_friday = True
include_easter_monday = True
include_christmas = True
include_boxing_day = True
```

**Comment by brunobord** [comment]
See linked PR that implements changes. It was hard work, but it made me clean up a lot of bizarre lines of code  :o)

**Comment by surfer190** [comment]
Thanks @brunobord everything looking good just made a small change for one of the labels

**Comment by brunobord** [comment]
closed by #286 & #287

**Comment by brunobord** [comment]
FYI, I've just released the v3.1.0 of workalendar that ships all the fixes. Happy upgrading!

* https://pypi.org/project/workalendar/3.1.0/
* https://github.com/peopledoc/workalendar/releases/tag/3.1.0
